File: gmao_backend/src/main/java/com/gmao/backend/entity/Zone.java
package com.gmao.backend.entity;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;
import com.fasterxml.jackson.annotation.JsonIgnore;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
@Entity
@Table(name = "zones")
public class Zone {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false)
    private String nom;

    @Enumerated(EnumType.STRING)
    private TypeZone type;

    @ManyToOne
    @JoinColumn(name = "batiment_id", nullable = false)
    @JsonIgnore
    private Batiment batiment;

    public enum TypeZone {
        ETAGE,
        SALLE,
        COULOIR,
        AUTRE
    }
}

File: gmao_mobile/lib/models/zone.dart
class Zone {
  final int id;
  final String nom;
  final String type; // ETAGE, SALLE, COULOIR, AUTRE
  final int batimentId;

  Zone({
    required this.id,
    required this.nom,
    required this.type,
    required this.batimentId,
  });

  factory Zone.fromJson(Map<String, dynamic> json) {
    return Zone(
      id: json['id'],
      nom: json['nom'] ?? 'Sans nom',
      type: json['type'] ?? 'AUTRE',
      batimentId: json['batiment'] != null ? json['batiment']['id'] : 0,
    );
  }
}
